docs/source/dataset.rst

```diff
@@ -0,0 +1,324 @@
+.. _dataset:
+
+Training Dataset Format
+=======================
+
+The Snips NLU library leverages machine learning algorithms and some training
+data in order to produce a powerful intent recognition engine.
+
+The better your training data is, and the more accurate your NLU engine will
+be. Thus, it is worth spending a bit of time to create a dataset that
+corresponds well to your use case.
+
+Snips NLU accepts two different dataset formats. The first one, which relies
+on YAML, is the preferred option if you want to create or edit a dataset
+manually.
+The other dataset format uses JSON and should rather be used if you plan to
+create or edit datasets programmatically.
+
+.. _yaml_format:
+
+===========
+YAML format
+===========
+
+The YAML dataset format allows you to define intents and entities using the
+`YAML <http://yaml.org/about.html>`_ syntax.
+
+.. _yaml_entity_format:
+
+Entity
+------
+
+Here is what an entity file looks like:
+
+.. code-block:: yaml
+
+    # City Entity
+    ---
+    type: entity # allows to differentiate between entities and intents files
+    name: city # name of the entity
+    values:
+      - london # single entity value
+      - [new york, big apple] # entity value with a synonym
```

Comment:
I would use YAML without brackets here. At first I got confused with the pseudo markdown format we use for utterances